lucille-plugins/lucille-tika/src/main/java/com/kmwllc/lucille/tika/stage/TextExtractor.java

```diff
@@ -103,14 +122,31 @@ public Iterator<Document> processDocument(Document doc) throws StageException {
       parseInputStream(doc, inputStream);
 
     } else if (doc.has(filePathField)) {
-
+      // get fileObject from path
       String filePath = doc.getString(filePathField);
+      FileObject file = FileUtils.getFileObject(filePath, fsManager);
+
+      // if file null, don't process document
+      if (file == null) {
+        return null;
+      }
+
+      // getting inputStream and catching errors gracefully
+      InputStream inputStream;
+      try {
+        inputStream = file.getContent().getInputStream();
+      } catch(FileSystemException e) {
+        log.warn("Error getting inputStream from file at: {}", filePath, e);
+        return null;
+      }
+
+      parseInputStream(doc, inputStream);
 
+      // close file after, do not need to check if file is null as we have already checked on line 130
       try {
-        InputStream inputStream = FileUtils.getInputStream(filePath);
-        parseInputStream(doc, inputStream);
-      } catch (IOException e) {
-        throw new StageException("InputStream cannot be parsed or created", e);
+        file.close();
```

Comment:
add comment saying parseInputStream would close the InputStream itself
add comment saying file.close() will close FileContent and InputStream as well?

Reply:
I suggested above creating the inputStream in a try-with-resources. The close() contract is a no-op if it is called multiple times. If some day parseInputStream is refactored and the implementation changes so that it doesn't close the inputStream (which would be a reasonable change IMO) then we at least know that the resource we created here is closed before we leave.